js/dispatch.ts

```diff
@@ -9,19 +9,31 @@ import { maybePushTrace } from "./trace";
 let nextCmdId = 0;
 const promiseTable = new Map<number, util.Resolvable<fbs.Base>>();
 
+let fireTimers: () => void;
+
+export function setFireTimersCallback(fn: () => void) {
+  fireTimers = fn;
+}
+
 export function handleAsyncMsgFromRust(ui8: Uint8Array) {
-  const bb = new flatbuffers.ByteBuffer(ui8);
-  const base = fbs.Base.getRootAsBase(bb);
-  const cmdId = base.cmdId();
-  const promise = promiseTable.get(cmdId);
-  util.assert(promise != null, `Expecting promise in table. ${cmdId}`);
-  promiseTable.delete(cmdId);
-  const err = errors.maybeError(base);
-  if (err != null) {
-    promise!.reject(err);
-  } else {
-    promise!.resolve(base);
+  // If a the buffer is empty, recv() on the native side timed out and we
+  // did not receive a message.
```

Comment:
ok so null received thru deno.recv() is quite special... I’m ok with this.

Reply:
It's a bit ugly, I agree, but I didn't want to add another callback for this.
Additionally, I think we will eventually pass *multiple* messages in a single buffer; in that light, treating an empty buffer as containing no messages at all seems quite natural.

Reply:
And the timer is special. I think it's okay.